src/wh_client.c

```diff
@@ -308,4 +311,206 @@ int wh_Client_CustomCbCheckRegistered(whClientContext* c, uint16_t id, int* resp
+    packet->keyCacheReq.sz = inSz;
+    /* set labelLen */
+    packet->keyCacheReq.labelSz = labelSz;
+    /* set id */
+    packet->keyCacheReq.id = keyId;
+    /* set label */
+    XMEMCPY(packet->keyCacheReq.label, label, labelSz);
+    /* write in */
+    XMEMCPY(packIn, in, inSz);
+    /* write request */
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_KEY, WH_KEY_CACHE,
+            WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->keyCacheReq) + inSz,
+            (uint8_t*)packet);
+}
+
+int wh_Client_KeyCacheRequest(whClientContext* c, uint32_t flags,
+    uint8_t* label, uint32_t labelSz, uint8_t* in, uint32_t inSz)
+{
+    return wh_Client_KeyCacheRequest_ex(c, flags, label, labelSz, in, inSz,
+        WOLFHSM_ID_ERASED);
+}
+
+int wh_Client_KeyCacheResponse(whClientContext* c, uint16_t* keyId)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == NULL)
+        return WH_ERROR_BADARGS;
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+        else
+            *keyId = packet->keyCacheRes.id;
+    }
+    return ret;
+}
+
+int wh_Client_KeyEvictRequest(whClientContext* c, uint16_t keyId)
+{
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == WOLFHSM_ID_ERASED)
+        return WH_ERROR_BADARGS;
+    /* set the keyId */
+    packet->keyEvictReq.id = keyId;
+    /* write request */
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_KEY, WH_KEY_EVICT,
+            WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->keyEvictReq),
+            (uint8_t*)packet);
+}
+
+int wh_Client_KeyEvictResponse(whClientContext* c)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    whPacket packet[1] = {0};
+    if (c == NULL)
+        return WH_ERROR_BADARGS;
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+    }
+    return ret;
+}
+
+int wh_Client_KeyExportRequest(whClientContext* c, uint16_t keyId)
+{
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == WOLFHSM_ID_ERASED)
+        return WH_ERROR_BADARGS;
+    /* set keyId */
+    packet->keyExportReq.id = keyId;
+    /* write request */
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_KEY, WH_KEY_EXPORT,
+            WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->keyExportReq),
+            (uint8_t*)packet);
+}
+
+int wh_Client_KeyExportResponse(whClientContext* c, uint8_t* label,
+    uint32_t labelSz, uint8_t* out, uint32_t* outSz)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    uint8_t rawPacket[WH_COMM_MTU] = {0};
+    whPacket* packet = (whPacket*)rawPacket;
+    uint8_t* packOut = (uint8_t*)(&packet->keyExportRes + 1);
+    if (c == NULL || labelSz > WOLFHSM_NVM_LABEL_LEN || outSz == NULL)
```

Comment:
labelSz == 1000000000000 would cause out of bounds read, we can't accept any labelSz

Reply:
I've thought more about this, we definitely need to validate labelSz because an incorrect value of 25 doesn't guarantee the buffer can actually handle `sizeof(metadata.label)` which i think is 24. if we have an unknown length we should bail out

Reply:
Scratch that, actually makes total sense for the buffer to be larger than max if they're reusing some generic buffer, my bad